babel/util.py

```diff
@@ -38,6 +38,9 @@ def distinct(iterable: Iterable[_T]) -> Generator[_T, None, None]:
 
     :param iterable: the iterable collection providing the data
     """
+    if isinstance(iterable, (tuple, list)) and len(iterable) <= 1:
+        yield from iterable
+        return
     seen = set()
     for item in iter(iterable):
```

Comment:
The itertools recipe suggests 

```suggestion
for item in filterfalse(seen.__contains__, iterable):
```

Another alternative would be:

```
yield from dict.fromkeys(iterable)
```

The language now guarantees that dictionaries preserve order, but this approach is non-iterable.

Reply:
`dict.fromkeys(iterable)` is clever! It's 5% faster than this in a microbenchmark.

What do you mean with "but this approach is non-iterable"?

Reply:
Yeah, we don't expect there to be very many items, and all of the internal uses are `list(distinct(...))` or `list(distinct(... + ...))` – TBH, I think we could micro-optimize that a bit too with an internal-use-only `_distinct_of(*iterables) -> list` function...

Reply:
See https://github.com/python-babel/babel/pull/1200/commits/72e25cf81f9542fb0b502527456d84cd14ef74cc :)